app/include/user_config.h

```diff
@@ -11,11 +11,13 @@
 // #define FLASH_16M
 #define FLASH_AUTOSIZE
 #define FLASH_SAFE_API
-// #define DEVELOP_VERSION
-#define FULL_VERSION_FOR_USER
 
-#define USE_OPTIMIZE_PRINTF
+// Byte 107 of esp_init_data_default, only one of these 3 can be picked
+#define ESP_INIT_DATA_ENABLE_READVDD33
+//#define ESP_INIT_DATA_ENABLE_READADC
```

Comment:
Hi thanks very much ,
I see i can use the Flash download tool and change the settings which 
effectively changes this setting.
from

#define ESP_INIT_DATA_ENABLE_READVDD33
//#define ESP_INIT_DATA_ENABLE_READADC

to

//#define ESP_INIT_DATA_ENABLE_READVDD33
#define ESP_INIT_DATA_ENABLE_READADC

I went the route yesterday afternoon of setting up Oracle VM ware and 
installing Linux increasing drive space and getting the firmware to 
compile ,
Changing a standard firmware source download to be the setting i wanted 
above in the user_config.h file and i compiled it and put the firmware 
on one of my units ,Surprisingly it worked ,

I would like to thank you although  I  haven't tried the change the bit 
at position 107 yet I will try it, just so I have 2 options ,
It seems that you can use a GUI and flag and burn it which seems much 
simpler than compiling the firmware from source.
I looked at the manual(3A_ESP8266_Flash_Tool_user_Manual_EN_v1.0.pdf) 
and on point 2.2 and it seems simple enough to change a flag , as well 
as other interesting settings i have not read about yet.

We should keep in touch.
regard JarrodMP





On 2016/02/06 16:08 PM, fixingthingsguy wrote:
>
> In app/include/user_config.h 
> <https://github.com/nodemcu/nodemcu-firmware/pull/746#discussion_r52102382>:
>
> >
> > -#define USE_OPTIMIZE_PRINTF
> > +// Byte 107 of esp_init_data_default, only one of these 3 can be picked
> > +#define ESP_INIT_DATA_ENABLE_READVDD33
> > +//#define ESP_INIT_DATA_ENABLE_READADC
> My experience with trying to build was extremely memorable, with an 
> unsuccessful outcome. I spent a lot of time  , and that's because I'm 
> no kind of expert. I finally gave up, there were too many gotcha's 
> with setting up Docker, VM, gitHub etc. Lots of hidden stuff an expert 
> would catch, but a novice doesn't even know to look for. Its best to 
> avoid that route.In my opinion, its far more preferable to stay with a 
> frightanic build and then use the esp tool to change the so-called 
> byte 107 to the way you want it.Based on expert suggestions I wrote 
> down what I did in #1007 
> <https://github.com/nodemcu/nodemcu-firmware/issues/1007>. It enables 
> me to stay consistent with the SDK and Dev/Master in one fell swoop 
> as you don't have to load the SDK 1.4  separately. If you already have 
> the right SDK, the notes in #911 
> <https://github.com/nodemcu/nodemcu-firmware/issues/911> walk you 
> through the byte 107 change.   From: Jarrod <notifications@github.com> 
> To: nodemcu/nodemcu-firmware <nodemcu-firmware@noreply.github.com> Cc: 
> fixingthingsguy <sbijanki@Yahoo.com> Sent: Friday, February 5, 2016 
> 11:18 PM Subject: Re: [nodemcu-firmware] Remove adc.readvdd33() 
> limitation. (#746 
> <https://github.com/nodemcu/nodemcu-firmware/pull/746>) In 
> app/include/user_config.h:>
> -#define USE_OPTIMIZE_PRINTF +// Byte 107 of esp_init_data_default, 
> only one of these 3 can be picked +#define 
> ESP_INIT_DATA_ENABLE_READVDD33 +//#define ESP_INIT_DATA_ENABLE_READADC
> I would like to build the firmware with this option , what or how is 
> the best root to take .— Reply to this email directly or view it on 
> GitHub.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/nodemcu/nodemcu-firmware/pull/746/files#r52102382>.
>
> No virus found in this message.
> Checked by AVG - www.avg.com <http://www.avg.com>
> Version: 2016.0.7357 / Virus Database: 4522/11566 - Release Date: 02/06/16
>